test/unit/x20schemapack.test.js

```diff
@@ -0,0 +1,230 @@
+    let folderPath = path.join(__dirname, '../data/swaggerMultifile/petstore-separate-yaml'),
+      files = [],
+      array = [
+        { fileName: folderPath + '/spec/swagger.yaml' },
+        { fileName: folderPath + '/spec/Pet.yaml' },
+        { fileName: folderPath + '/spec/parameters.yaml' },
+        { fileName: folderPath + '/spec/NewPet.yaml' },
+        { fileName: folderPath + '/common/Error.yaml' }
+      ];
+
+    array.forEach((item) => {
+      files.push({
+        content: fs.readFileSync(item.fileName, 'utf8'),
+        fileName: item.fileName
+      });
+    });
+
+    var schema = new SchemaPack({ type: 'folder', data: files });
+    schema.mergeAndValidate((err, status) => {
+      if (err) {
+        expect.fail(null, null, err);
+      }
+      if (status.result) {
+        schema.convert((error, result) => {
+          if (error) {
+            expect.fail(null, null, err);
+          }
+          expect(result.result).to.equal(true);
+          expect(result.output.length).to.equal(1);
+          expect(result.output[0].type).to.have.equal('collection');
+          expect(result.output[0].data).to.have.property('info');
+          expect(result.output[0].data).to.have.property('item');
+          done();
+        });
+      }
+      else {
+        expect.fail(null, null, status.reason);
+        done();
+      }
+    });
+  });
+
+  it('Should merge correctly the files in folder - basicExample', function(done) {
+    let folderPath = path.join(__dirname, '../data/swaggerMultifile/uberTest'),
+      files = [],
+      array = [
+        { fileName: folderPath + '/index.yaml' },
+        { fileName: folderPath + '/definitions/Activities.yaml' },
+        { fileName: folderPath + '/definitions/Activity.yaml' },
+        { fileName: folderPath + '/definitions/Error.yaml' },
+        { fileName: folderPath + '/definitions/Product.yaml' },
+        { fileName: folderPath + '/definitions/ProductList.yaml' },
+        { fileName: folderPath + '/definitions/Profile.yaml' },
+        { fileName: folderPath + '/definitions/PriceEstimate.yaml' }
+      ];
+
+    array.forEach((item) => {
+      files.push({
+        content: fs.readFileSync(item.fileName, 'utf8'),
+        fileName: item.fileName
+      });
+    });
+
+    var schema = new SchemaPack({ type: 'folder', data: files });
+    schema.mergeAndValidate((err, status) => {
+      if (err) {
+        expect.fail(null, null, err);
+      }
+      if (status.result) {
+        schema.convert((error, result) => {
+          if (error) {
+            expect.fail(null, null, err);
+          }
+          expect(result.result).to.equal(true);
+          expect(result.output.length).to.equal(1);
+          expect(result.output[0].type).to.have.equal('collection');
+          expect(result.output[0].data).to.have.property('info');
+          expect(result.output[0].data).to.have.property('item');
+          done();
```

Comment:
We should be also checking for data that's coming from other files instead of very basic info that's available in all converted collection.